Inital commit for Hamster Time Tracker ⏰

rabin-io opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

hfiguiere commented on org.gnome.Hamster.json:
```diff
@@ -0,0 +1,71 @@
+{
+    "app-id": "org.gnome.Hamster",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "40",
```
```suggestion
    "runtime-version": "41",
```

rabin-io replied:
changed to 41

hfiguiere commented on org.gnome.Hamster.json:
```diff
@@ -0,0 +1,71 @@
+                {
+                        "type": "archive",
+                        "url": "https://github.com/projecthamster/hamster/archive/refs/tags/v3.0.2.tar.gz",
+                        "dest-filename": "hamster-3.0.2.tar.gz",
```
```suggestion
```

rabin-io replied:
removed

hfiguiere commented on org.gnome.Hamster.GUI.desktop:
```diff
@@ -0,0 +1,20 @@
+[Desktop Entry]
```
desktop file is upstream, why adding it here?

rabin-io replied:
removed.

hfiguiere commented:
bot, build org.gnome.Hamster

hfiguiere commented:
There is a mismatch between the app-id in the upstream metainfo and this manifest. But the upstream metainfo is wrong given the DBus registration. (the app-id is used to automatically grant DBus permissions)

rabin-io commented:
What do you recommend doing?

hfiguiere commented:
patching it (and submit upstream) and use the `rename-appdata-file` to rename it in the flatpak manifest. Also the desktop file need renaming.

rabin-io commented:
Hi @rabin-io 

I'm sorry, I'm not so versed in the subject, can you be more specific what needs to change and where?

barthalion commented on org.gnome.Hamster.json:
```diff
@@ -0,0 +1,70 @@
+        "--socket=wayland",
+        "--socket=fallback-x11",
+        "--filesystem=xdg-documents",
+        "--own-name=org.gnome.Hamster",
+        "--own-name=org.gnome.Hamster.GUI",
+        "--own-name=org.gnome.Hamster.WindowServer"
```
I may be wrong about it, but AFAIK Flatpak allows anything matching `${APPID}.*` by default.

rabin-io replied:
How can we validate that ?

hfiguiere replied:
remove them a run it. You should get debug errors if that's a problem.

rabin-io replied:
I removed this lines, and it seems to work as well.

barthalion commented on org.gnome.Hamster.json:
```diff
@@ -0,0 +1,70 @@
+    ],
+    "finish-args": [
+        "--socket=wayland",
+        "--socket=fallback-x11",
```
```suggestion
        "--socket=fallback-x11",
        "--share=ipc",
```

rabin-io replied:
added

barthalion commented on org.gnome.Hamster.json:
```diff
@@ -0,0 +1,70 @@
+            ]
+        }
+    ],
+    "finish-args": [
```
`finish-args` are usually places above `modules`

rabin-io replied:
I reordered the sections.

rabin-io commented:
Also, should it convert the file to yaml ? 
seems to be much more readable.

nedrichards commented:
Up to you on json or yaml.

nedrichards commented:
bot, build org.gnome.Hamster

nedrichards commented on org.gnome.Hamster.json:
```diff
@@ -0,0 +1,69 @@
+            "sources": [
+                {
+                        "type": "archive",
+                        "url": "https://github.com/rabin-io/hamster/archive/refs/tags/release-3.0.2.tar.gz",
```
Sorry, this may already have been answered but is there a reason this doesn't point to the upstream source release but to your fork instead?

nedrichards commented on org.gnome.Hamster.json:
```diff
@@ -0,0 +1,69 @@
+        "--socket=fallback-x11",
+        "--share=ipc",
+        "--filesystem=xdg-documents",
+        "--own-name=org.gnome.Hamster",
```
```suggestion
        "--own-name=org.gnome.Hamster"
```
This should fix the build.

rabin-io commented:
Not sure how i did it, but I closed the PR, and I can't reopen it. 

Also...

- I converted the file to yaml
- and rebased over the `new-pr` branch (was over master)

Can you change the PR to be against new-pr instead of master ?
or should I go back to way it was before ?

hfiguiere commented:
It's probably faster to resubmit a new PR. We can take it from there.

rabin-io commented:
> It's probably faster to resubmit a new PR. We can take it from there.

Opened new PR
